### PR TITLE
Mcnp6 ptrac

### PR DESCRIPTION
**Pull request by MigMagan**
Compatibility fix: small changes to `mcnp.py` to make it compatible with MCNP6-generated ptrac files. Also added an MCNP6 ptrac file in tests,  generated running MCNP6 with the same `mcnp_ptrac_inp.txt` .

### CODE REVIEW

**Comment by MigMagan** [comment]
No idea why this would be failing. It is related to nuc_data, but I haven't touched that?

```
Testing test_data_checksums.py:

.E
======================================================================
ERROR: test_data_checksums.test_internal_hashes
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/root/repo/tests/test_data_checksums.py", line 48, in test_internal_hashes
    hashtools.set_internal_hashes(test_data)
  File "/root/.local/lib/python3.6/site-packages/pyne/dbgen/hashtools.py", line 52, in set_internal_hashes
    set_hash(item, nuc_data)
  File "/root/.local/lib/python3.6/site-packages/pyne/dbgen/hashtools.py", line 118, in set_hash
    the_hash = calc_hash(node, nuc_data)
  File "/root/.local/lib/python3.6/site-packages/pyne/dbgen/hashtools.py", line 90, in calc_hash
    if type(item[:]) == numpy.ndarray:
  File "/usr/local/lib/python3.6/dist-packages/tables/group.py", line 439, in __getitem__
    return self._f_get_child(childname)
  File "/usr/local/lib/python3.6/dist-packages/tables/group.py", line 708, in _f_get_child
    self._g_check_has_child(childname)
  File "/usr/local/lib/python3.6/dist-packages/tables/group.py", line 391, in _g_check_has_child
    node_type = self._g_get_objinfo(name)
  File "tables/hdf5extension.pyx", line 986, in tables.hdf5extension.Group._g_get_objinfo
AttributeError: 'slice' object has no attribute 'encode'
-------------------- >> begin captured logging << --------------------
numexpr.utils: INFO: Note: NumExpr detected 36 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.
numexpr.utils: INFO: NumExpr defaulting to 8 threads.
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
Ran 2 tests in 0.045s

FAILED (errors=1)
Testing failed on test_data_checksums.py
```

Could it be related to the changes in https://github.com/pyne/data/commit/fd999360f2aa18cf6fee11d785e2dbd248bb6b01 ?

**Comment by bam241** [comment]
I believe it is deeper than that... (cc @gonuke)

I think we broke the hashtools with my/our MaterialLibrary update…
it is expecting a to open material_library as a group and it is a dataset…

Now that we have updated `nuc_data.h5` to the new format, then we see than problem.

I'll have a PR to fix this soon
I keep you posted

**Comment by bam241** [comment]
issued a fix in #1337 . Hopefully it will fix this.

**Comment by MigMagan** [comment]
Excuse my ignorance, but, how should I proceed now? Do I wait for #1337 to be merged or just pull the commits from your repo?

**Comment by bam241** [comment]
sadly one  need to wait for #1337 to be merged.
Most of the active reviewer seems to be busy at the time...
 
If you need to use branch I don't believe this error will be blocking... it should only impact the tests ...

File: pyne/mcnp.py [hunk 1]
```diff
@@ -1204,8 +1204,13 @@ def read_variable_ids(self):
...
         if self.eightbytes:
-            variable_info = self.read_next(
-                "qqqqqqqqqqqiiiiiiiii", 124, raw_format=True)
+            mcnp_version=self.mcnp_version_info.split()[1]
+            if mcnp_version in ["6","6.mpi"]:
+                variable_info = self.read_next(
+                    "iqqqqqqqqqqiiiiiiiii", 120, raw_format=True)
+            else: # Not sure about MCNPX 
+                variable_info = self.read_next(
+                    "qqqqqqqqqqqiiiiiiiii", 124, raw_format=True)
```
Comment (bam241) [inline pyne/mcnp.py:1215]: I am not sure I understand this change.
Could you add a comment to explain briefly why `mcnp6` is not read the same way as the other `mcnp` version ?
(i.e. why `iqqqqqqqqqqiiiiiiiii, 120` in one case and `qqqqqqqqqqqiiiiiiiii, 124` in the other)

**Review by bam241 — APPROVED** [review]
this looks good to me.

As I don't understand it, I just would like a small comment to explain the change in the `read_next` call for `mcnp6`.

File: pyne/mcnp.py (hunk 1 above)
Comment (MigMagan) [reply pyne/mcnp.py:1215]: Well,  the thing here is that the variable MCNP uses in `ptrak()` for the first value in that register is explicitly declared as `i8knd` in MCNP5, and as `integer` in MCNP6, thus the change in format and length. 

However, on second thought, it is possible to compile MCNP6 with 8 byte integers by default, IIRC, so I may have to try different compile options and test again...

**Comment by bam241** [comment]
@MigMagan rebasing should fix the tests.

File: pyne/mcnp.py (hunk 1 above)
Comment (MigMagan) [reply pyne/mcnp.py:1215]: The 8 bit integers in MCNP6 is quite a can of worms, as even the first registry is no longer what we expect. I am adding an exception to handle the situation and calling it a day. If there is interest in it (i.e: users actually using such compilation options), support can always be added later.

**Comment by MigMagan** [comment]
Guess #1319 is hitting the checks...

**Review by gonuke — APPROVED** [review]
This looks good to me. Thanks @MigMagan

**Pull request merged**